### ISSUE TITLE
[BOUNTY]: $55 BayStation-esque "Clientside" Musicplayer

### ISSUE THREAD

**Issue by Tsar-Salat** [BeeStation/BeeStation-Hornet#11268]
## Overview

I'd like it if Baystation's clientside musicplayer was ported.

The musicplayer should replace Lobbymusic with a 'Track System" that displays the title of the song, author, url, and license. If a different track is selected, or the current one ends, the new track should display all of its information to the chatbox.

A list of resources, if needed:
- https://github.com/Baystation12/Baystation12/pull/25152
- https://github.com/Baystation12/Baystation12/pull/30346

Example:

![image](https://github.com/user-attachments/assets/d36a35b7-f108-432e-8b18-8ca255cec9bd)

A button to change the current track, potentially in the OOC tab:

![image](https://github.com/user-attachments/assets/f2cd2611-516a-45ef-b70e-a8c846cfc362)

### Code Preview
<details>
<summary>Code</summary>

Baystation has structured it like this

![image](https://github.com/user-attachments/assets/bf1f21fb-8907-48c5-9334-a150e8c1a46e)

Important licensing structure

![image](https://github.com/user-attachments/assets/211355a6-53ba-44eb-810e-97b9bb551013)

</details>

## tl;dr

Overall:
- The Lobby music system as-is should be gutted entirely.
- A track should play upon a client joining the lobby.
- When a track plays, all pertinent song information should appear as a `to_chat` or equivalent to the client
- There should be a uniform & datumized structure for adding new songs

<details>
<summary>**_Optional $20 Bonus: work the musicplayer into the current disco dance machine, akin to Baystation's Jukebox._**</summary>

![image](https://github.com/user-attachments/assets/a9af9258-1610-49f4-83f5-0ba98a401acf)

</details>

## Payment

Payment methods I can do: Ko-Fi, Patreon, Steam game(s) with total value equivalent to 65 USD. 

I'd like to avoid PayPal, but if you twist my arm I'd probably still accept it.

**Comment by PowerfulBacon** [comment]
Have thought about client-side music player before with the jukebox, not sure how it would work with regards to how the audio source moves as you move around although tbf byond doesn't actually properly handle that anyway

**Comment by PowerfulBacon** [comment]
Clientside stereo panning is currently impossible, but would become possible when WebView2 is released. Volume panning would work though if we wanted clientside jukeboxes which would be nice to prevent freezing when you walk in range of a jukebox

**Comment by PowerfulBacon** [comment]
Doing some research the following things are possible:
- Lobby music could be changed to be based on URL, rather than on a sound-file and played through the TGchat music system (which would make it appear as if it was admin music).
- TGUI Chat could have support added for music playing in the world, with the volume scaling as your camera moves around. This would only support a single music peice playing at any time without support for multiple audio players, however this would probably be pretty janky. I would code it so that when you enter the range of a new song, the previously playing one would fade out and the new one would fade in.
- Music would need a priority, so that global music overwrites music playing locally.

**Comment by PowerfulBacon** [comment]
https://github.com/user-attachments/assets/cc271b04-7f4b-43e9-95c1-3570c84fa719

This is a basic implementation of positional TGUI panel audio/web-sounds, needs expanding but is showing promise so far. The jukebox was added to make it look cool, its actually just a verb which creates a sound at your mobs location for debugging purposes.

**Comment by PowerfulBacon** [comment]
ah shit, you wanted a button for the client themselves to change the song rather than the whole server to have synced music. I'll see what I can do about that.

**Comment by Tsar-Salat** [comment]
Unstale, a pr is open
